rest/RESTClient.ts

```diff
@@ -63,8 +72,17 @@ export default class RESTClient {
         const discordBucket = api.headersIn['x-ratelimit-bucket'] as string | undefined;
         if (api.headersIn['x-ratelimit-global'] === 'true') {
           this.globallyRateLimited = true;
+          this.queue.push({ method, endpoint: endpointFinal, auth, payload });
+
           setTimeout(() => {
             this.globallyRateLimited = false;
+
+            while (this.queue.length >= 1) {
+              const _request = this.queue.shift();
+
+              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+              this.request(_request!.method, _request!.endpoint, _request!.auth, _request!.payload);
```

Comment:
I know I asked you to change this so this probably doesn't matter but the non null assertions aren't necessary anyways

Reply:
And this